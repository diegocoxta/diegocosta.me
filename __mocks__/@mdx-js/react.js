module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
