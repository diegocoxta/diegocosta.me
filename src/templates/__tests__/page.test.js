import React from 'react';
import { render } from '@testing-library/react';

jest.mock('../../components/Header', () => () => <p>Header</p>);
jest.mock('../../components/Footer', () => () => <p>Footer</p>);
jest.mock('../../components/Metatags', () => () => <p>Metatags</p>);

import PageTemplate from '../page';

describe('<PageTemplate>', () => {
  it('renders properly', () => {
    const props = {
      data: {
        page: {
          frontmatter: {
            title: 'Title 1',
          },
          fields: {
            slug: '/path-to-article',
          },
          body: 'Post content',
        },
      },
    };

    const { baseElement, getByText } = render(<PageTemplate {...props} />);
    expect(getByText('Post content')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
