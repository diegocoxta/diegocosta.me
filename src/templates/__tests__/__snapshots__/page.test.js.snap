// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PagesTemplate> renders properly 1`] = `
.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 40px 0 0 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  margin: 0 5px;
  color: #2F3D4F;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 22px;
}

.c3 {
  background: linear-gradient(90deg,#ffffff 20px,transparent 1%) center, linear-gradient(#ffffff 20px,transparent 1%) center, #2F3D4F;
  background-size: 22px 22px;
  height: 120px;
  margin: 40px 0;
  width: 100%;
}

.c6 {
  color: #2F3D4F;
  box-shadow: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c6:hover,
.c6:focus {
  border-bottom: 1px solid #2F3D4F;
  outline: none;
}

.c4 {
  margin: 0 0 60px 0;
  font-weight: 400;
  color: #2F3D4F;
}

.c5 {
  margin: 20px 0 0 0;
}

.c7 {
  line-height: 1.5;
  font-size: 19px;
}

.c7 a {
  color: #AAC13D;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid #ffffff;
}

.c7 a:hover,
.c7 a:focus {
  border-bottom: 1px solid #AAC13D;
  outline: none;
}

.c7 blockquote {
  border-left: 5px solid #AAC13D;
  padding-left: 20px;
  margin-left: 0;
}

.c7 img {
  box-shadow: none !important;
}

.c7 h3 {
  font-size: 28px;
}

.c7 table.uses-table {
  width: 100%;
  margin: 40px 0;
}

.c7 table.uses-table td {
  text-align: center;
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <a
            aria-label="languageswitcher.toggle languages.pt"
            class="c2"
            href="/about-us"
            language="pt"
          >
            /
            pt
          </a>
          <a
            aria-label="languageswitcher.toggle languages.en"
            class="c2"
            href="/about-us"
            language="en"
          >
            en
          </a>
          <a
            aria-label="languageswitcher.toggle languages.es"
            class="c2"
            href="/about-us"
            language="es"
          >
            es
          </a>
        </div>
      </main>
    </section>
    <p>
      Header
    </p>
    <main>
      <p>
        Metatags
      </p>
      <div
        class="c3"
      />
      <section
        class="c0"
      >
        <main>
          <article
            class="c4"
            data-testid="article-item"
          >
            <header
              class="c5"
            >
              <a
                class="c6"
                data-testid="article-header-title"
                href="/path-to-article/"
              >
                Title 1
              </a>
            </header>
            <section
              class="c7"
            >
              Post content
            </section>
          </article>
        </main>
      </section>
    </main>
    <p>
      Footer
    </p>
  </div>
</body>
`;
